packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -90,6 +94,23 @@ export class CdkToolkit {
         throw new Error(`Stack ${stack.name} does not define an environment, and AWS credentials could not be obtained from standard locations or no region was configured.`);
       }
 
+      if (Object.keys(stack.template.Resources || {}).length === 0) { // The generated stack has no resources
+        const cfn = await options.sdk.cloudFormation(stack.environment.account, stack.environment.region, Mode.ForReading);
+        if (!await stackExists(cfn, stack.name)) {
+          warning('%s: stack has no resources, skipping deployment.', colors.bold(stack.name));
+        } else {
+          warning('%s: stack has no resources, skipping deployment and deleting existing stack.', colors.bold(stack.name));
+          await this.destroy({
+            stackNames: [stack.name],
+            exclusively: false,
+            force: false,
```

Comment:
If you don't pass `force: true`, the user will be presented with a prompt asking if they're sure they want to destroy the stack. As our current flow here is what could be seen as a "regular" deployment, we shouldn't be prompting users.

Reply:
Updated to set `force` to `true` but this means that if for *any reason* the synthesized stack has no resource (it could be a "human error") the existing stack is automatically deleted, are you comfortable with this?

Reply:
That's no different than if the stack includes only one resource by accident...

Reply:
You are right